### PR TITLE
Use apollo onError in with-apollo example

### PR DESCRIPTION
**Pull request by fabienheureux**
Use Apollo onError in order to flag GraphQL errors in the next.js server logs in the with-apollo example
https://www.apollographql.com/docs/react/api/link/apollo-link-error/

## Documentation / Examples

- [x] Make sure the linting passes

~⚠️ I lazily created this PR from Github directly as I don't have a local build of next.js on my current machine. 
If ever linting does not pass, I will update this later today (sorry for the noise). 
EDIT: it does not, I will update this PR shortly.~

### CODE REVIEW

**Comment by fabienheureux** [comment]
any chance this can be reviewed ?

**Review by styfle — APPROVED** [review]

**Comment by fabienheureux** [comment]
@ijjk any change I can get another approval from you or any code owner ? 
Thanks !

**Review by ijjk — APPROVED** [review]

**Pull request merged**
